### PR TITLE
[Rule Tuning] MacOS Installer Package Net Event

### PR DESCRIPTION
**Pull request by DefSecSentinel**
## Issues

https://github.com/elastic/detection-rules/issues/2189

## Summary

This rule logic is flawed. I spent yesterday afternoon executing every installer package payload [Mystikal](https://github.com/D00MFist/Mystikal) has to offer along with various other malicious [packages](https://redcanary.com/blog/clipping-silver-sparrows-wings/) and this rule did not fire once. The problem with the rule is in the way the process tree spawns with installer packages preventing sequencing on process.entity_id as the child processes spawned from installer or package_script_service are not linked directly by process id and therefore cannot be connected together via the process.entity_id. This will prevent the rule from ever firing.

### CODE REVIEW

**Review by Mikaayenson — APPROVED** [review]
Great work! This will also make the RTA much easier to create.

File: rules/macos/execution_installer_package_spawned_network_event.toml [hunk 1]
```diff
@@ -19,29 +20,24 @@ from = "now-9m"
...
 query = '''
-sequence by process.entity_id with maxspan=1m
-  [process where event.type == "start" and host.os.family == "macos" and
-    process.parent.executable in ("/usr/sbin/installer", "/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer") ]
-  [network where not cidrmatch(destination.ip,
-    "10.0.0.0/8", "127.0.0.0/8", "169.254.0.0/16", "172.16.0.0/12", "192.0.0.0/24", "192.0.0.0/29", "192.0.0.8/32",
-    "192.0.0.9/32", "192.0.0.10/32", "192.0.0.170/32", "192.0.0.171/32", "192.0.2.0/24", "192.31.196.0/24",
-    "192.52.193.0/24", "192.168.0.0/16", "192.88.99.0/24", "224.0.0.0/4", "100.64.0.0/10", "192.175.48.0/24",
-    "198.18.0.0/15", "198.51.100.0/24", "203.0.113.0/24", "240.0.0.0/4", "::1", "FE80::/10", "FF00::/8")]
+sequence by host.id, user.id with maxspan=30s
```
Comment (DefSecSentinel) [inline rules/macos/execution_installer_package_spawned_network_event.toml:36]:
```suggestion
sequence by host.id, user.id with maxspan=1m
```

**Review by terrancedejesus — APPROVED** [review]

File: rules/macos/execution_installer_package_spawned_network_event.toml [hunk 2]
```diff
@@ -19,29 +20,24 @@ from = "now-9m"
...
-  [process where event.type == "start" and host.os.family == "macos" and
-    process.parent.executable in ("/usr/sbin/installer", "/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer") ]
-  [network where not cidrmatch(destination.ip,
-    "10.0.0.0/8", "127.0.0.0/8", "169.254.0.0/16", "172.16.0.0/12", "192.0.0.0/24", "192.0.0.0/29", "192.0.0.8/32",
-    "192.0.0.9/32", "192.0.0.10/32", "192.0.0.170/32", "192.0.0.171/32", "192.0.2.0/24", "192.31.196.0/24",
-    "192.52.193.0/24", "192.168.0.0/16", "192.88.99.0/24", "224.0.0.0/4", "100.64.0.0/10", "192.175.48.0/24",
-    "198.18.0.0/15", "198.51.100.0/24", "203.0.113.0/24", "240.0.0.0/4", "::1", "FE80::/10", "FF00::/8")]
+sequence by host.id, user.id with maxspan=1m
+[process where event.type == "start" and process.parent.name : ("installer", "package_script_service") and process.name : ("bash", "sh", "zsh", "python", "osascript", "tclsh*")] 
+[network where event.type == "start" and process.name : ("curl", "osascript", "wget", "python")]
```
Comment (DefSecSentinel) [inline rules/macos/execution_installer_package_spawned_network_event.toml:38]:
```suggestion
sequence by host.id, user.id with maxspan=30s
[process where event.type == "start" and event.action == "exec" and process.parent.name : ("installer", "package_script_service") and process.name : ("bash", "sh", "zsh", "python", "osascript", "tclsh*")] 
[network where event.type == "start" and process.name : ("curl", "osascript", "wget", "python")] 
```

**Review by imays11 — APPROVED** [review]
I changed a typo in the description. But looks good

**Pull request merged**